spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensor.scala

```diff
@@ -866,6 +866,34 @@ private[tensor] class DenseTensor[@specialized T: ClassTag](
     (values, indices)
   }
 
+  protected def sumNSquare(n: T): T = {
+    var res: T = ev.fromType(0)
+    val func = new TensorFunc2[T] {
+      override def apply(data1: Array[T], offset1: Int): Unit = {
+        res = ev.plus(res, ev.pow(ev.abs(data1(offset1)), n))
+      }
+    }
+    DenseTensorApply.apply1[T](this, func)
```

Comment:
Instead of using apply1, how about using sth like vsMul?

Reply:
Updated to use MKL